src/test/java/com/commercetools/sync/inventory/InventorySyncTest.java

```diff
@@ -0,0 +1,214 @@
+                Channel.referenceOfId(KEY_3));
+        final List<InventoryEntryDraft> toProcess = new ArrayList<>(drafts);
+        toProcess.add(draftWithNewChannel);
+        final InventorySync inventorySync = new InventorySync(InventorySyncOptionsBuilder.of(mockCtpClient())
+                .build(), mockThrowingInventoryService(), mock(TypeService.class));
+
+        inventorySync.syncDrafts(toProcess);
+        final InventorySyncStatistics stats = inventorySync.getStatistics();
+        assertThat(stats.getProcessed()).isEqualTo(10);
+        assertThat(stats.getFailed()).isEqualTo(7);
+    }
+
+    private InventorySync getInventorySyncer(int batchSize, int parallelThreads, boolean ensureChannels) {
+        final InventorySyncOptions options = InventorySyncOptionsBuilder.of(mockCtpClient())
+                .setBatchSize(batchSize)
+                .setParallelProcessing(parallelThreads)
+                .ensureChannels(ensureChannels)
+                .build();
+        return new InventorySync(options, mockInventoryService(), mock(TypeService.class));
+    }
+
+    private CtpClient mockCtpClient() {
+        final CtpClient ctpClient = mock(CtpClient.class);
+        final BlockingSphereClient client = mock(BlockingSphereClient.class);
+        when(ctpClient.getClientConfig()).thenReturn(SphereClientConfig.of(CTP_KEY, CTP_ID, CTP_SECRET));
+        when(ctpClient.getClient()).thenReturn(client);
+        return ctpClient;
+    }
+
+    private InventoryService mockInventoryService() {
+        final Channel mockedChannel = mockChannel(KEY_3, REF_3);
+        final InventoryService inventoryService = mock(InventoryService.class);
+        when(inventoryService.fetchAllSupplyChannels()).thenReturn(channelsDB);
+        when(inventoryService.fetchInventoryEntriesBySkus(any())).thenReturn(inventoriesDB);
+        when(inventoryService.createSupplyChannel(any())).thenReturn(mockedChannel);
+        when(inventoryService.createInventoryEntry(any())).thenReturn(null);
+        when(inventoryService.updateInventoryEntry(any(), any())).thenReturn(null);
+        return inventoryService;
+    }
+
+    /**
+     *
+     * @return mock of {@link InventoryService} that throws {@link RuntimeException} on creating and updating calls
+     */
+    private InventoryService mockThrowingInventoryService() {
+        final InventoryService inventoryService = mock(InventoryService.class);
+        when(inventoryService.fetchAllSupplyChannels()).thenReturn(channelsDB);
+        when(inventoryService.fetchInventoryEntriesBySkus(any())).thenReturn(inventoriesDB);
+        when(inventoryService.createSupplyChannel(any())).thenThrow(new RuntimeException());
+        when(inventoryService.createInventoryEntry(any())).thenThrow(new RuntimeException());
+        when(inventoryService.updateInventoryEntry(any(), any())).thenThrow(new RuntimeException());
+        return inventoryService;
+    }
+
+    private Channel mockChannel(String id, String key) {
```

Comment:
I would separate these util functions in a InventorySyncMockUtils class so they can be reused and to avoid this class getting bigger